api/src/main/resources/moodle.yaml

```diff
@@ -0,0 +1,172 @@
+#This file provide the end points for moodle interactions
+# MOodle API information can be found at :
+# https://docs.moodle.org/dev/Web_service_API_functions
+
+swagger: '2.0'
+
+info:
+  title: Moodle
+  description: |
+    Service for Moodle Interactions.
+  version: "0.1"
+
+host: "aoudev.nnlm.gov"
+schemes:
+  - "https"
+basePath: /webservice/rest
+
+produces:
+  - application/json
+
+##########################################################################################
+## PATHS
+##########################################################################################
+
+paths:
+  /server.php:
+    get:
+      summary: Retrieves user information from Moodle.
+      description: "Get user moodle information"
+      tags:
+        - Moodle
+      consumes:
+        - application/json
+      operationId: getMoodleId
+      parameters:
+        - name: wsfunction
+          in: query
+          description: Moodle component method name
+          type: string
+          required: true
+        - name: moodlewsrestformat
+          in: query
+          description: Format of the response
+          type: string
+          required: true
+        - name: field
+          in: query
+          description: Input field eg email on the basis of which information is received
+          type: string
+          required: false
+        - name: values[0]
+          in: query
+          description: Value of the input field
+          type: string
+          required: false
+        - name: wstoken
+          in: query
+          description: Token
+          type: string
+          required: true
+      responses:
+        200:
+          description: Response received
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/MoodleUserResponse'
+        400:
+          description: Bad request
+        500:
+          description: Moodle Internal Error
+      security:
+        - AdminSecurity: []
+
+  /server.php?wsfunction=core_badges_get_user_badges:
```

Comment:
I tried doing that but then the method was being overridden because of which i started using server.php and another one with argument. 

Swagger 3.0 has i think 'OneOf' keyword that helps in scenarios like this where the response can be different for a path.